src/main/java/io/jenkins/plugins/coverage/adapter/XMLCoverageReportAdapter.java

```diff
@@ -36,22 +36,27 @@ public XMLCoverageReportAdapter(String path) {
      * @param source source xml file
      */
     @Override
-    public Document convert(File source) {
+    public Document convert(File source) throws ConversionException {
         File xsl;
         try {
             xsl = getRealXSL();
         } catch (FileNotFoundException e) {
             e.printStackTrace();
             throw new ConversionException(e);
         }
-        return XMLUtils.getInstance().convertToDocumentWithXSL(xsl, source);
+        try {
+            return XMLUtils.getInstance().convertToDocumentWithXSL(xsl, source);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            throw new ConversionException(e);
+        }
     }
 
     @SuppressWarnings("WeakerAccess")
-    protected File getRealXSL() throws FileNotFoundException {
+    protected File getRealXSL() throws ConversionException, FileNotFoundException {
         try {
             File realXSL = new File(getClass().getResource(getXSL()).toURI());
```

Comment:
`getXSL()` is marked as Nullable, but you do not check it here. I would suggest annotationg `getXSD` as `@CheckForNull` instead of Nullable. It would enforce FindBugs to check it and fail the build if it happens.

In the case of this method likely it needs to return null in this case as well

Reply:
I have marked `getXSL()` with `@CheckForNull`. Do you mean that I should check the return value of `getXSL()` in `getRealXSL()`?

Reply:
yes.

Reply:
I understand. Thanks~